### PR TITLE
Added Keychron V10

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Added new keyboard V10 which contains two sub versions.
Thanks!

## Description

Added ANSI and ISO versions with  encoder.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by keychron-dev** [comment]
@tzarc can you help check this? thanks.

File: keyboards/keychron/v10/v10.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_LED_INDEX, 255, 255, 255);
+    } else {
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_LED_INDEX, 0, 0, 0);
+        }
+    }
+    return true;
+}
+
+#endif // CAPS_LOCK_LED_INDEX
```
Comment (drashna) [inline keyboards/keychron/v10/v10.c:86]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#endif // CAPS_LOCK_LED_INDEX

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/v10/v10.c (hunk 1 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/v10/v10.c:86]: All right, thanks~

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/v10/ansi_encoder/info.json [hunk 2]
```diff
@@ -0,0 +1,111 @@
+{
+    "keyboard_name": "Keychron V10",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x01A1",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/v10/ansi_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/v10/ansi_encoder/rules.mk [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/v10/ansi_encoder/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/v10/iso_encoder/info.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
+{
+    "keyboard_name": "Keychron V10",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x03A3",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/v10/iso_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/v10/iso_encoder/rules.mk [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/v10/iso_encoder/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/v10/readme.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+```base
+make keychron/v10/v10_ansi_stm32l432:default
+```
+
+Flashing example for this keyboard:
+
+```base
+make keychron/v10/v10_ansi_stm32l432:default:flash
+```
```
Comment (fauxpark) [inline keyboards/keychron/v10/readme.md:19]:
```suggestion
    make keychron/v10/ansi_encoder:default
    make keychron/v10/iso_encoder:default

Flashing example for this keyboard:

    make keychron/v10/ansi_encoder:default:flash
    make keychron/v10/iso_encoder:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/v10/readme.md (hunk 6 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/v10/readme.md]: Thanks~

**Comment by keychron-dev** [comment]
@fauxpark check this one too if you have time, thanks

**Comment by keychron-dev** [comment]
@fauxpark hi, do you have time to check? sorry to bother you again.

**Review by fauxpark — APPROVED** [review]

**Comment by keychron-dev** [comment]
@drashna  can you help merge this? it has two approvals already.

**Pull request merged**